include/util/struct_packing.hpp

```diff
@@ -0,0 +1,19 @@
+#ifndef STRUCT_PACKING_HPP
+#define STRUCT_PACKING_HPP
+
+/*
+    Macro to pack struct/class in both environment
+    Usage:  PACK( 
+        struct MyStruct 
+        { 
+            int Property1;
+        });
+*/
+
+#ifdef _MSC_VER
+#define PACK( input ) __pragma( pack(push, 1) ) input __pragma( pack(pop) )
```

Comment:
I found that wen using the pragma version, gcc would not throw warnings if the struct was not packable (at least on 6.1) - I'm not sure if the pragma just does the packing switch anyway or if this would result in undefined behavior. Hence the switch to __attribute which complains about fishy stuff correctly in gcc. Maybe we should have a compiler specific switch in the define?